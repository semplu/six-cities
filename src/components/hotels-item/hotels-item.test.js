import React from 'react';
import renderer from 'react-test-renderer';
import HotelsItem from './hotels-item';

it(`HotelsItem correctly renders after launch`, () => {
  const tree = renderer.create(<HotelsItem
    key={0}
    id={0}
    name={`test_name`}
    price={0}
    type={`test_type`}
    image={`test_image_url`}
    status={`test_status`}
    rating={0}
    link={`test_link`}
    goToHotel={jest.fn()}
    addActiveHotel={jest.fn()}
    removeActiveHotel={jest.fn()}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
