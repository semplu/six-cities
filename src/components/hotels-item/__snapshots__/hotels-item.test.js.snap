// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelsItem correctly renders after launch 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[MockFunction]}
>
  <div
    className="place-card__mark"
  >
    <span>
      test_status
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="test_link"
    >
      <img
        alt="test_name"
        className="place-card__image"
        height="200"
        src="test_image_url"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          0
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
      onClick={[MockFunction]}
    >
      <a
        href="test_link"
      >
        test_name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      test_type
    </p>
  </div>
</article>
`;
